packages/components/src/core/NavigationHeader/components/NavigationHeaderSecondaryNav/index.tsx

```diff
@@ -39,26 +50,98 @@ export interface NavigationHeaderSecondaryNavProps {
   items: NavigationHeaderSecondaryNavItem[];
   hasInvertedStyle?: boolean;
   isNarrow?: boolean;
+  sdsStyle?: "dropdown" | "drawer";
+  expandedAccordion?: string | null;
+  setExpandedAccordion?: (value: string | null) => void;
+  accordionRefs?: React.MutableRefObject<Map<string, HTMLDivElement>>;
+  scrollToAccordion?: (accordionId: string) => void;
 }
 
 export default function NavigationHeaderSecondaryNav({
   menuProps,
   items,
   hasInvertedStyle,
   isNarrow,
+  sdsStyle = "dropdown",
+  expandedAccordion: expandedAccordionProp,
+  setExpandedAccordion: setExpandedAccordionProp,
+  accordionRefs: accordionRefsProp,
+  scrollToAccordion,
 }: NavigationHeaderSecondaryNavProps) {
   const [anchorEl, setAnchorEl] = useState<Element | null>(null);
   const [activeDropdownKey, setActiveDropdownKey] = useState<string | null>(
     null
   );
+  const [activeDrawerKey, setActiveDrawerKey] = useState<string | null>(null);
   const theme: SDSTheme = useTheme();
+  const hoverTimeoutRef = useRef<NodeJS.Timeout | null>(null);
+  const [contentKey, setContentKey] = useState<string | null>(null);
+  const [isContentFading, setIsContentFading] = useState(false);
+
+  // Use props if provided, otherwise use local state (for backward compatibility)
+  const [localExpandedAccordion, setLocalExpandedAccordion] = useState<
+    string | null
+  >(null);
+  const localAccordionRefs = useRef<Map<string, HTMLDivElement>>(new Map());
+
+  const expandedAccordion = expandedAccordionProp ?? localExpandedAccordion;
+  const setExpandedAccordion =
+    setExpandedAccordionProp ?? setLocalExpandedAccordion;
+  const accordionRefs = accordionRefsProp ?? localAccordionRefs;
 
   const open = Boolean(anchorEl);
+  const drawerOpen = activeDrawerKey !== null;
+
   function onClose() {
     setAnchorEl(null);
     setActiveDropdownKey(null);
   }
 
+  function onDrawerClose() {
```

Comment:
definitely not for this PR, but I wonder if the PrimaryNav and SecondaryNav could share some common code to keep things DRY?

Reply:
Yeah, I actually tried to extract some of the code into a shared component, but I’ll make sure to refactor it further later to reduce redundancy. Thanks for the tip! :D